lib/pc_patch.c

```diff
@@ -608,3 +608,117 @@ PCPOINT *pc_patch_pointn(const PCPATCH *patch, int n)
 	pcerror("%s: unsupported compression %d requested", __func__, patch->type);
 	return NULL;
 }
+
+
+/**
+* Function used by pc_patch_set_schema. Copy the data of p1 in p2. Assume that
+* p1 and p2 schemas are compatible, meaning that the dimensions in p2 that are
+* also in p1 have the same interpretations. The value "defaults" is used for
+* dimensions that are not in p1.
+*/
+static void
+pc_patch_point_copy(const PCPOINT *p1, PCDIMENSION **dims,
+		double defaults, PCPOINT *p2)
+{
+	size_t i;
+	for ( i = 0; i < p2->schema->ndims; i++ )
+	{
+		PCDIMENSION *d1 = dims[i];
+		PCDIMENSION *d2 = p2->schema->dims[i];
+
+		if ( d1 )
+		{
+			uint8_t *data2 = p2->data + d2->byteoffset;
+			uint8_t *data1 = p1->data + d1->byteoffset;
+			memcpy(data2, data1, d1->size);
+		}
+		else
+		{
+			pc_point_set_double(p2, d2, defaults);
+		}
+	}
+}
+
```

Comment:
I propose the slightly lower level version, to use a default buffer rather than reinterpret default dimensions each time. The input point is only used for its payload, so I find it more obvious to pass only its data buffer :
```
void pc_point_set(PCPOINT *p, const uint8_t *data, const PCDIMENSION **dims, const uin8_t *defaults)
{
	size_t i;
	for ( i = 0; i < p->schema->ndims; i++ )
	{
		PCDIMENSION *ddim = dims[i];
		PCDIMENSION *pdim = p->schema->dims[i];
		uint8_t *pdata = p->data + pdim->byteoffset;
		uint8_t *ddata = ddim ? (data + ddim->byteoffset) : (defaults + pdim->byteoffset);
		memcpy(pdata, ddata, pdim->size);
	}
}
```